### PR TITLE
Fix User Data Collection checkbox

### PR DESCRIPTION
**Pull request by mlassak**
Resolves: #354 

## Description
Replaced the default ```null``` state of User Data Collection checkbox. Additionally, fixed a potential cause of exceptions due to missing null checks in ```clusterSettingsUtils.ts``` when dashboard config notebook controller spec is not present.

## How Has This Been Tested?
Tested locally

## Merge criteria:
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ const DEFAULT_IDLENESS_CHECK_PERIOD = '1'; // 1 minute
 const DEFAULT_CLUSTER_SETTINGS: ClusterSettings = {
   pvcSize: DEFAULT_PVC_SIZE,
   cullerTimeout: DEFAULT_CULLER_TIMEOUT,
-  userTrackingEnabled: null,
+  userTrackingEnabled: false,
 };
 
 export const updateClusterSettings = async (
```
Comment (andrewballantyne) [inline backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts:20]: This method is a bit of a mess. Things you need to change:

1. Avoid adding more try/catches for single API calls... just do a `.catch` on the end and let it flow through to the next call
2. Avoid throwing errors when it doesn't matter if the call succeeded; if the call is not critical to the success of the API call, don't break threads
3. If you do not first check if the K8s resource exists, doing patches/deletes and the likewise will be counter productive -- look to follow step no1 and just do `.catch` and suppress those failures
4. If you're going to catch for 404, don't forget to throw the error -- or return -- if it's NOT 404 -- see the bottom catch of this method, it just defaults to `undefined` being returned from this function; which Fastify does not like and it breaks the API call from the client

If we were not moving away from the server with the pass-through API, I'd request this function be broken up into several other functions so we can do proper separation of concerns. The saving of this page saves multiple items and it's hard to follow which is which inside this function -- makes it harder to maintain and non-related items can cause problems (which is what the original bug noted).

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
This didn't entirely solve the bug. If you lack some of the resources used in the `updateClusterSettings` call, it breaks the API call (doesn't end it, just fails to ever complete). This can lead to issues with the UI since the requests browsers make are limited concurrently and thus you can block all other API calls if the user spams the button.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
I suspect your cluster has resources that mine does not.

We should be able to gracefully handle the lack of resources. Otherwise we need to present this info upfront to the user that they do not have a proper configuration. I think in all cases in this `clusterSettingsUtils`, we can gracefully handle 404s.

My error is (`e.body.message`): `message: 'configmaps "notebook-controller-culler-config" not found',` 404 status code.

Which looks to be the scenario `my culling is disabled, delete configmap` -- definitely don't need to crash and burn on that call. (line 51)

**Comment by maroroman** [comment]
Sorry @andrewballantyne did not mean that re-review yet

**Review by andrewballantyne — APPROVED** [review]
hmmm... we don't save it (if you lack the files) and say we saved it 😞 My catch-all logic sorta made the call do nothing, but look like it succeeded. Gosh, I really don't want to go overboard with refactoring this function 🤔 

Okay... so this is a pre-existing problem. We fixed my concern about it not updating properly and crashing when things went off the rails. Let's go with this, and we can work to improve this method. Ideally we take this to the client shortly after #279
